libraries/AP_Logger/AP_Logger.cpp

```diff
@@ -1170,6 +1170,113 @@ AP_Logger::log_write_fmt *AP_Logger::msg_fmt_for_name(const char *name, const ch
     return f;
 }
 
+// as above but does not assume pointers will stick around.
+AP_Logger::log_write_fmt *AP_Logger::msg_fmt_for_name_scripting(const char *name, const char *labels, const char *units, const char *mults, const char *fmt)
+{
+    WITH_SEMAPHORE(log_write_fmts_sem);
+    struct log_write_fmt *f;
+    for (f = log_write_fmts; f; f=f->next) {
+        // direct comparison used from scripting where pointer is not maintained
+        if (strcmp(f->name,name) == 0) {
```

Comment:
Now this is a different function we could keep a separate list of scripting formats, this should significantly speed up scripting logging as we would only have to do one or two string compares.

Reply:
Could also just up none scripting formats at the end of the linked list....